src/main/java/com/liefery/android/stop_badge/StopBadgeFactory.java

```diff
@@ -0,0 +1,38 @@
+package com.liefery.android.stop_badge;
+
+import android.annotation.SuppressLint;
+import android.graphics.*;
+import android.util.Log;
+import android.util.LruCache;
+
+@SuppressLint( "NewApi" )
+public class StopBadgeFactory {
+
+    private static StopBadgeFactory instance = null;
+
+    private BadgeCache cache = new BadgeCache();
+
+    protected StopBadgeFactory() {
+        // Exists only to defeat instantiation.
+    }
+
+    public static Bitmap getBitmap( StopBadge badge, int size ) {
+        if ( instance == null ) {
+            instance = new StopBadgeFactory();
+        }
+
+        if ( instance.cache.get( badge.toKey( size ) ) == null ) {
+            Bitmap newBadge = badge.export( size );
+            instance.cache.put( badge.toKey( size ), newBadge );
+            return newBadge;
+        } else {
+            return instance.cache.get( badge.toKey( size ) );
+        }
+    }
+
+    public class BadgeCache extends LruCache<String, Bitmap> {
+        BadgeCache() {
+            super( 2 * 1024 * 1024 ); // 2MiB
```

Comment:
This is the output for generating a StopBadge without any changes for `100px` to `800px`
```
Bitmap b1: 100px 40000bytes //0.04MB
D  Bitmap b2: 200px 160000bytes
D  Bitmap b3: 300px 360000bytes
D  Bitmap b4: 400px 640000bytes
D  Bitmap b5: 500px 1000000bytes
D  Bitmap b6: 600px 1440000bytes
D  Bitmap b7: 700px 1960000bytes
D  Bitmap b8: 800px 2560000bytes //2.56 MB
```